File: ViewModels/DriverDashboardViewModel.swift
//
//  DriverDashboardViewModel.swift
//  UDrive
//
//  Created on 2024
//

import Foundation

class DriverDashboardViewModel: ObservableObject {
    @Published var availableRides: [Ride] = []
    @Published var activeRides: [Ride] = []
    @Published var isOnline: Bool = false
    
    private let mockService = MockDataService.shared
    
    func loadRides(for driverId: UUID) {
        activeRides = mockService.rides.filter { $0.driverId == driverId && ($0.status == .accepted || $0.status == .inProgress) }
        availableRides = mockService.rides.filter { $0.status == .requested && $0.driverId == nil }
    }
    
    func toggleOnlineStatus() {
        isOnline.toggle()
    }
    
    func acceptRide(_ ride: Ride) {
        if let index = mockService.rides.firstIndex(where: { $0.id == ride.id }) {
            mockService.rides[index].driverId = UUID() // In real app, use current driver ID
            mockService.rides[index].status = .accepted
            loadRides(for: UUID()) // In real app, use current driver ID
        }
    }
}

File: lib/viewmodels/driver_dashboard_viewmodel.dart
import 'package:flutter/foundation.dart';
import '../services/mock_data_service.dart';
import '../models/ride.dart';

class DriverDashboardViewModel extends ChangeNotifier {
  final MockDataService _mockService = MockDataService();
  List<Ride> _availableRides = [];
  List<Ride> _activeRides = [];
  bool _isOnline = false;

  List<Ride> get availableRides => List.unmodifiable(_availableRides);
  List<Ride> get activeRides => List.unmodifiable(_activeRides);
  bool get isOnline => _isOnline;

  void loadRides(String driverId) {
    _activeRides = _mockService.rides
        .where((ride) =>
            ride.driverId == driverId &&
            (ride.status == RideStatus.accepted ||
                ride.status == RideStatus.inProgress))
        .toList();
    _availableRides = _mockService.rides
        .where((ride) =>
            ride.status == RideStatus.requested && ride.driverId == null)
        .toList();
    notifyListeners();
  }

  void toggleOnlineStatus() {
    _isOnline = !_isOnline;
    notifyListeners();
  }

  void acceptRide(Ride ride) {
    // In real app, this would update the ride status
    loadRides('2'); // Mock driver ID
  }
}

